src/base/bittorrent/session.cpp

```diff
@@ -321,6 +321,11 @@ QString Session::tempPath() const
     return m_tempPath;
 }
 
+QString Session::torrentTempPath(const InfoHash &hash) const
+{
+    return m_tempPath + QString(hash) + "/";
```

Comment:
`QString(hash)` is 40 chars right? isn't that a bit too long, or this shouldn't be concerned of?
most filesystem has a upper limit of 255 bytes (more or less).